astropy/table/table.py

```diff
@@ -4092,116 +4101,93 @@ def to_pandas(self, index=None, use_nullable_int=True):
           2002-01-01  2.0    6.0     8.0 0 days 00:03:20
 
         """
-        from pandas import DataFrame, Series
+        from .dataframes import to_pandas
 
-        if index is not False:
-            if index in (None, True):
-                # Default is to use the table primary key if available and a single column
-                if self.primary_key and len(self.primary_key) == 1:
-                    index = self.primary_key[0]
-                else:
-                    index = False
-            else:
-                if index not in self.colnames:
-                    raise ValueError(
-                        "index must be None, False, True or a table column name"
-                    )
-
-        def _encode_mixins(tbl):
-            """Encode a Table ``tbl`` that may have mixin columns to a Table with only
-            astropy Columns + appropriate meta-data to allow subsequent decoding.
-            """
-            from astropy.time import TimeBase, TimeDelta
-
-            from . import serialize
-
-            # Convert any Time or TimeDelta columns and pay attention to masking
-            time_cols = [col for col in tbl.itercols() if isinstance(col, TimeBase)]
-            if time_cols:
-                # Make a light copy of table and clear any indices
-                new_cols = []
-                for col in tbl.itercols():
-                    new_col = (
-                        col_copy(col, copy_indices=False) if col.info.indices else col
-                    )
-                    new_cols.append(new_col)
-                tbl = tbl.__class__(new_cols, copy=False)
-
-                # Certain subclasses (e.g. TimeSeries) may generate new indices on
-                # table creation, so make sure there are no indices on the table.
-                for col in tbl.itercols():
-                    col.info.indices.clear()
-
-                for col in time_cols:
-                    if isinstance(col, TimeDelta):
-                        # Convert to nanoseconds (matches astropy datetime64 support)
-                        new_col = (col.sec * 1e9).astype("timedelta64[ns]")
-                        nat = np.timedelta64("NaT")
-                    else:
-                        new_col = col.datetime64.copy()
-                        nat = np.datetime64("NaT")
-                    if col.masked:
-                        new_col[col.mask] = nat
-                    tbl[col.info.name] = new_col
+        return to_pandas(self, index=index, use_nullable_int=use_nullable_int)
 
-            # Convert the table to one with no mixins, only Column objects.
-            encode_tbl = serialize.represent_mixins_as_columns(tbl)
-            return encode_tbl
+    def to_df(
+        self,
+        backend: str | types.ModuleType,
+        index: bool | str | None = None,
+        use_nullable_int: bool = True,
```

Comment:
Since this is brand new API and we don't need to worry about backward compatibility yet, I would suggest making *all* arguments either positional-only *or* keyword-only.
positional-or-keyword-allowed arguments (the default), are extremely hard to deprecate, move around, or rename, if we ever need to.

```suggestion
        backend: str | types.ModuleType,
        /,
        *,
        index: bool | str | None = None,
        use_nullable_int: bool = True,
```

Here, I'm suggesting making `backend` positional-only, since it seems to match your prefered style (according to doc examples), and everything else keyword-only; this is an arbitrary suggestion based on the common idiom that arguments that have default values are often *meant* to be passed as keywords, even though this is rarely enforced. `use_nullable_int` being a boolean flag strongly suggests that it *should* never be allowed as positional.

Reply:
I also don't think there's much point in allowing multiple types for the `backend` argument; users can already pass `module.__name__` if they really need to pass modules dynamically, and I'm worried that making the interface flexible for such a small gain might reveal itself as a bad decision in a few years.

I do understand that we're just re-exposing existing type-flexibility from `nw.Implementation.from_backend(backend)`, but I don't see the point of doing so in our public API (though it's fine in internal functions, which we are allowed to change without notice)